thundra/integrations/botocore.py

```diff
@@ -683,6 +684,67 @@ def before_call(self, scope, wrapped, instance, args, kwargs, response, exceptio
         scope.span.set_tag(constants.AwsSDKTags['SERVICE_NAME'], service_name)
 
 
+class AWSStepFunctionIntegration(BaseIntegration):
+    CLASS_TYPE = 'sfn'
+
+    def __init__(self):
+        pass
+
+    def get_operation_name(self, wrapped, instance, args, kwargs):
+        _, request_data = args
+        state_machine_arn = request_data.get('stateMachineArn')
+        if state_machine_arn:
+            return state_machine_arn.split(':')[-1]
+        return constants.AWS_SERVICE_REQUEST
+
+    def before_call(self, scope, wrapped, instance, args, kwargs, response, exception):
+        scope.span.domain_name = constants.DomainNames['AWS']
+        scope.span.class_name = constants.ClassNames['STEPFUNCTIONS']
+
+        _, request_data = args
+        state_machine_arn = request_data.get('stateMachineArn', '')
+        execution_name = request_data.get('name', '')
+
+        service_name = instance.__class__.__name__.lower()
+
+        try:
+            orig_input = request_data.get('input', None)
+            if orig_input != None:
+                parsed_input = json.loads(orig_input)
+                trace_link = str(uuid.uuid4())
+                parsed_input['_thundra'] = {
+                    "trace_link": trace_link,
+                    "step": 0
+                }
+                scope.span.set_tag(constants.AwsStepFunctionsTags['EXECUTION_INPUT'], orig_input)
+                request_data['input'] = json.dumps(parsed_input)
+                scope.span.set_tag(constants.SpanTags['TRACE_LINKS'], [trace_link])
+                scope.span.set_tag(constants.SpanTags['RESOURCE_TRACE_LINKS'], [trace_link])
```

Comment:
Can we add this temporary info into span without tag (for ex. a direct attribute)? Because when this temporary information is added as tag, it is send to our backend and indexed. To overcome this problem, we can drop this tag after processing but it complicates the logic I believe. So my suggestion is put this info as an attribute to span and set it here only for AWS Step Function calls